### DISCUSSION TITLE
About struct redefinition (C) 구조체 재정의

### DISCUSSION THREAD

**Discussion by gyomin0315** [community/community#47180] [Programming Help] [answered]
### Select Topic Area

Question

### Body

#include <stdio.h>

struct content {

	char content[40];
	int price;
	double rate;
};

int main(void)
{
	struct content c1;
	struct content c1 = {0};
	struct content c2 = c1;
	struct content c3 = { 0 }; // 싹 다 0으로 초기화 해줌.

	c3 = c1;

	printf("c1 = %s %d %.1lf\n",c1.content, c1.price, c1.rate);
	printf("c2 = %s %d %.1lf\n", c2.content, c2.price, c2.rate);
	printf("c3 = %s %d %.1lf\n", c3.content, c3.price, c3.rate);
	

}

Above program can't be compiled because content c1 is redefined. By initializing c1, I could solve this problem.
But i don't know why struct can't be redefined.
Your answers will be a lot help to me.

**Comment by jge162** [comment, marked as answer]
you are declaring c1 twice, you will get an error for that. Also, because c1 is declared wrong c2 = c1 should fail because you trying to redeclare c1 again under c2 = c1.

**Reply by jge162** [reply]
here is an example 
```cpp
#include <stdio.h>

struct content {

char content[40];
int price;
double rate;
};

int main(void)
{
  struct content c1 = {"item1", 100, 10.5};
  struct content c2;
  
  for (int i = 0; i < sizeof(c1) / sizeof(c1.content[0]); i++) {
    c2.content[i] = c1.content[i];
  }
  c2.price = c1.price;
  c2.rate = c1.rate;

  printf("c1 = %s %d %.1lf\n",c1.content, c1.price, c1.rate);
  printf("c2 = %s %d %.1lf\n", c2.content, c2.price, c2.rate);
}
```

**Reply by gyomin0315** [reply]
Thank you. Your answer helped me a lot.

**Reply by jricciardi** [reply]
I've gone ahead and marked this as the answer - thanks for jumping in @jge162!
